### ISSUE TITLE
Show POI icon in red if closed (or hide their icons)

### ISSUE THREAD

**Issue by raindropsfromsky** [osmandapp/Osmand#3350]
When we are looking for a POI category nearby, the idea is to go there immediately and use its service.

But often a restaurant or store has service hours, and it may be closed when we look at the map.
There is no point in considering such "closed right now!" POIs in the map.

But some of the POIs may re-open by the time the user reaches there, so the user may like to check them out for such possibility.

Therefore show their icon in red.

**Comment by knusprjg** [comment]
I came here to propose the same thing :) Additionally I'd color the currently open POIs in green. Otherwise you'd have to look at each to see if the opening hours are there at all.

**Comment by hakuchi** [comment]
I think there can be a **small dot** in one corner of the POI symbol.
If the POI is **open** the dot can be **green**.
If the POI is **closed** the dot can change to **red**.
If there are no opening hours in the OSM database there should be no dot displayed.

Changing the entire color of a POI to green is not good, because supermarket is pink, fuel station is blue, sport is green and restaurants are brown.

**Comment by vshcherb** [comment]
Just to add to original issue: nowadays you could filter out open/closed POI via search.

**Comment by raindropsfromsky** [comment]
I like @hakuchi's idea as compared to my original suggestion: Let the original icon's natural color be seen, and let there be an overlay symbol to show open/close status.

@vshcherb, yes, I know there's the filter, but then we get to see the open and closed POIs separately. Therefore the overlay idea is better.

**Comment by KOLANICH** [comment]
>If the POI is closed the dot can change to red.

And if a POI is closing soon the color is interpolated through yellow and orange. If a POI is opening soon the color is interpolated through pink and blue.

**Comment by matkoniecz** [comment]
> Just to add to original issue: nowadays you could filter out open/closed POI via search.

How it can be done? I missed option to do it like in Goggle Maps (after search buttons appear with "show open", "show top-rated" etc)

**Comment by HolgerJeromin** [comment]
> How it can be done?

It is the icon on the right top:
![rps20190704_151630_075](https://user-images.githubusercontent.com/2410353/60669495-bf06dd00-9e6e-11e9-93b8-71aa57bbd839.jpg)

**Comment by matkoniecz** [comment]
Thanks for the hint! Though, unlike Google Maps there is no reasonable way to enable it once in a map view.

**Comment by Murmele** [comment]
> > How it can be done?
> 
> It is the icon on the right top: ![rps20190704_151630_075](https://user-images.githubusercontent.com/2410353/60669495-bf06dd00-9e6e-11e9-93b8-71aa57bbd839.jpg)

I don't see the settingsbutton in the right upper corner. Is it only on IOS?

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/92decd70-5261-4ad2-966a-ec6d0a22ebee"  width="300" height="auto"/> </th>
  </tr>
</table>

OsmAnd 5.1.3, veröffentlicht: 2025-06-20

**Comment by sonora** [comment]
@Murmele Dont't do a freetext search (like 'supermarket'), use the predefined category 'Convenience store and supermarket' instead. Looks like the filter screen is tailored per category, and there seems no filter icon for an arbitrary freetext search.

**Comment by Murmele** [comment]
> [@Murmele](https://github.com/Murmele) Dont't do a freetext search (like 'supermarket'), use the predefined category 'Convenience store and supermarket' instead. Looks like the filter screen is tailored per category, and there seems no filter icon for an arbitrary freetext search.

Thanks that worked well!
It is a little bit inconsistent. Would be nice to have the filtering also for the freetext search

**Comment by sonora** [comment]
Yes, true. I guess the question is what filter items to put on, because free text search hits could be a mixture of _anything_,, so it's hard to decide which properties would make sense to offer for filtering in the first place.

**Comment by Murmele** [comment]
This is true, but at least the "currently open" filtering would be an improvement
